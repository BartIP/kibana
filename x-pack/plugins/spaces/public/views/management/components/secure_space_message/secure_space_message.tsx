/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { EuiLink, EuiSpacer, EuiText } from '@elastic/eui';
<<<<<<< HEAD
=======
import { FormattedMessage } from '@kbn/i18n/react';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import { UserProfile } from 'plugins/xpack_main/services/user_profile';
import React, { Fragment } from 'react';

interface Props {
  userProfile: UserProfile;
}

export const SecureSpaceMessage = (props: Props) => {
  if (props.userProfile.hasCapability('manageSecurity')) {
    return (
      <Fragment>
        <EuiSpacer />
        <EuiText className="eui-textCenter">
          <p>
<<<<<<< HEAD
            Want to assign a role to a space? Go to Management and select{' '}
            <EuiLink href="#/management/security/roles">Roles</EuiLink>.
=======
            <FormattedMessage
              id="xpack.spaces.management.secureSpaceMessage.howToAssignRoleToSpaceDescription"
              defaultMessage="Want to assign a role to a space? Go to Management and select {rolesLink}."
              values={{
                rolesLink: (
                  <EuiLink href="#/management/security/roles">
                    <FormattedMessage
                      id="xpack.spaces.management.secureSpaceMessage.rolesLinkText"
                      defaultMessage="Roles"
                    />
                  </EuiLink>
                ),
              }}
            />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
          </p>
        </EuiText>
      </Fragment>
    );
  }
  return null;
};
