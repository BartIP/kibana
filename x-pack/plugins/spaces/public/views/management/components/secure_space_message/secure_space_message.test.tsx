/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */
<<<<<<< HEAD
import { shallow } from 'enzyme';
import React from 'react';
=======
import React from 'react';
import { shallowWithIntl } from 'test_utils/enzyme_helpers';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import { SecureSpaceMessage } from './secure_space_message';

describe('SecureSpaceMessage', () => {
  it(`doesn't render if user profile does not allow security to be managed`, () => {
    const userProfile = {
      hasCapability: (key: string) => {
        if (key === 'manageSecurity') {
          return false;
        }
        throw new Error(`unexpected capability ${key}`);
      },
    };

<<<<<<< HEAD
    expect(shallow(<SecureSpaceMessage userProfile={userProfile} />)).toMatchSnapshot();
=======
    expect(shallowWithIntl(<SecureSpaceMessage userProfile={userProfile} />)).toMatchSnapshot();
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  });

  it(`renders if user profile allows security to be managed`, () => {
    const userProfile = {
      hasCapability: (key: string) => {
        if (key === 'manageSecurity') {
          return true;
        }
        throw new Error(`unexpected capability ${key}`);
      },
    };

<<<<<<< HEAD
    expect(shallow(<SecureSpaceMessage userProfile={userProfile} />)).toMatchSnapshot();
=======
    expect(shallowWithIntl(<SecureSpaceMessage userProfile={userProfile} />)).toMatchSnapshot();
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  });
});
