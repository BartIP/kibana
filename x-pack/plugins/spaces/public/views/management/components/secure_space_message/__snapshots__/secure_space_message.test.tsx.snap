// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecureSpaceMessage doesn't render if user profile does not allow security to be managed 1`] = `""`;

exports[`SecureSpaceMessage renders if user profile allows security to be managed 1`] = `
<React.Fragment>
  <EuiSpacer
    size="l"
  />
  <EuiText
    className="eui-textCenter"
    grow={true}
    size="m"
  >
    <p>
<<<<<<< HEAD
      Want to assign a role to a space? Go to Management and select
       
      <EuiLink
        color="primary"
        href="#/management/security/roles"
        type="button"
      >
        Roles
      </EuiLink>
      .
=======
      <FormattedMessage
        defaultMessage="Want to assign a role to a space? Go to Management and select {rolesLink}."
        id="xpack.spaces.management.secureSpaceMessage.howToAssignRoleToSpaceDescription"
        values={
          Object {
            "rolesLink": <EuiLink
              color="primary"
              href="#/management/security/roles"
              type="button"
            >
              <FormattedMessage
                defaultMessage="Roles"
                id="xpack.spaces.management.secureSpaceMessage.rolesLinkText"
                values={Object {}}
              />
            </EuiLink>,
          }
        }
      />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
    </p>
  </EuiText>
</React.Fragment>
`;
